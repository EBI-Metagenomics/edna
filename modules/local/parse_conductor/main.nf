
process PARSE_CONDUCTOR {
    
    // Parse the trimming_conductor output and store 
    // the flags into environment variables 
    tag "$meta.id"
    label 'very_light'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        "https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:${params.mpt_version}":
        "biocontainers/mgnify-pipelines-toolkit:${params.mpt_version}" }"

    input:
    tuple val(meta), path(trimming_conductor_out)

    output:
    tuple val(meta), env(fwd_flag), env(rev_flag), emit: conductor_out
    path "versions.yml"                          , emit: versions

    script:
    """
    fwd_flag=\$(sed '1q;d' "${trimming_conductor_out}")
    rev_flag=\$(sed '2q;d' "${trimming_conductor_out}")

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: ${params.mpt_version}
    END_VERSIONS
    """
}
