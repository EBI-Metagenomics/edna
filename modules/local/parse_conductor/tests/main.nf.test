nextflow_process {

    name "Test Process PARSE_CONDUCTOR"
    script "../main.nf"
    process "PARSE_CONDUCTOR"
    tag "modules"
    tag "parse_conductor"

    test("Should read the flag file from trimming_conductor and turn the decisions into environment variables") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('${projectDir}/modules/local/parse_conductor/tests/data/test_trimming_conductor_out.txt', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.conductor_out).match("conductor_out") }
            )
        }

    }

}
