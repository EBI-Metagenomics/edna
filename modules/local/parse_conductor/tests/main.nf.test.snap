{
    "conductor_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "std",
                    "std"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-24T16:26:52.07037968"
    }
}